tests/test_dictionary.c

```diff
@@ -0,0 +1,269 @@
+    char *s = (char*)malloc(sizeof(char) * 3);
+    s = "no";
+    check_add_to_dict(NULL, s, 1);
+}
+
+/* Test adding a long string for success (not in dict) */
+Test(dictionary, add_to_dict_s1) {
+    char *s = (char*)malloc(sizeof(char) * 53);
+    s = "nojfkdsjfksdjfdsjfkdsjfjfkdsjfkdsjfkdsjkfiejkdjskfsd";
+    check_add_to_dict(NULL, s, 1);
+}
+
+/*
+************ read_to_dict tests ****************************
+*/
+
+/* Test arg 1 for null input in read_to_dict */
+Test(dictionary, read_to_dict_null0) {
+    dict_t *d;
+    int rc;
+
+    d = dict_new();
+
+    rc = read_to_dict(NULL, d);
+
+    cr_assert_eq(rc, 0, "Passing NULL in as a string in read_to_dict should have "
+                 "returned 0, but it returned %d", rc);
+}
+
+/* Test arg 2 for null input in read_to_dict */
+Test(dictionary, read_to_dict_null1) {
+    int rc;
+
+    rc = read_to_dict("./tests/dict_test0.txt", NULL);
+
+    cr_assert_eq(rc, -1, "Passing NULL in as a dictionary in read_to_dict should have "
+                 "returned -1, but it returned %d", rc);
+}
+
+/* Test bad filename input in read_to_dict */
+Test(dictionary, read_to_dict_noname) {
+    dict_t *d;
+    int rc;
+
+    d = dict_new();
+
+    rc = read_to_dict("DNE.txt", d);
+
+    cr_assert_eq(rc, 0, "Passing NULL in as a string in read_to_dict should have "
+                 "returned 0, but it returned %d", rc);
+}
+
+void check_read_to_dict(char *file, int expected) {
+    dict_t *d;
+    int rc;
+
+    d = dict_new();
+
+    rc = read_to_dict(file, d);
+
+    cr_assert_eq(rc, 1, "Passing file %s in read_to_dict should have "
+                 "returned %d, but it returned %d", file, expected, rc);
+}
+
+/* Test for regular input in read_to_dict 1 */
+Test(dictionary, read_to_dict0) {
+    check_read_to_dict("./tests/dict_test0.txt", 1);
+}
+
+
+
+/*
+************ in_dict tests ****************************
+*/
+
+/* Test arg 1 for null input for in_dict */
+Test(dictionary, in_dict_null0) {
+    dict_t *d;
+    int rc;
+
+    d = dict_new();
+
+    rc = in_dict(NULL, d);
+
+    cr_assert_eq(rc, -1, "Passing NULL in as a string in in_dict should have "
+                 "returned -1, but it returned %d", rc);
+}
+
+/* Test arg 2 for null input for in_dict */
+Test(dictionary, in_dict_null1) {
+    int rc;
+
+    rc = in_dict("hi", NULL);
+
+    cr_assert_eq(rc, -1, "Passing NULL in as a dictionary in in_dict should have "
+                 "returned -1, but it returned %d", rc);
+}
+
+/* Test empty string input for in_dict */
+Test(dictionary, in_dict_empty) {
+    dict_t *d;
+    int rc;
+
+    d = dict_new();
+
+    read_to_dict("./tests/dict_test0.txt", d);
+
+    rc = in_dict("", d);
+
+    cr_assert_eq(rc, -1, "Passing an empty string in as a dictionary to in_dict "
+                 "should have returned -1, but it returned %d", rc);
+}
+
+/* Helper function for testing regular input for in_dict */
+void check_in_dict(char *file, char *str, int expected) {
```

Comment:
Using `int`s as `bool`s is so 154. You have `stdbool.h` imported; use it. Make `expected` a `bool` and change all the callers to use `true` and `false` - it makes the callers much more readable.

Reply:
Since the functions have been updated to return EXIT_SUCCESS and EXIT_FAILURE as appropriate, is it ok to change the checking functions to compare to EXIT_SUCCESS and EXIT_FAILURE as opposed to true and false? 

I think true would correspond to EXIT_FAILURE and false would correspond to EXIT_SUCCESS otherwise.